### PR TITLE
Problem: configs are baked in testground

### PR DESCRIPTION
**Pull request by mmsqe**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Summary by CodeRabbit

- **New Features**
	- Introduced a new command-line argument `--num_max_workers` for enhanced control over concurrency during testing.
	- Added a new configuration parameter `num_max_workers` in the benchmark settings for improved transaction processing flexibility.
	- Enhanced load generation functionality to utilize the specified maximum number of workers.
	- Improved timestamp handling and transaction performance metrics in benchmarking reports.
	- Added new methods for calculating transactions per second (TPS) and truncating fractional seconds in timestamp reporting.
	- Enhanced configuration handling in the `gen` function for more granular control over node and application settings.

- **Bug Fixes**
	- Improved handling of concurrency parameters in various benchmark functions for better performance and resource management.

### CODE REVIEW

File: testground/README.md [hunk 1]
```diff
@@ -85,13 +85,15 @@ $ nix run github:crypto-org-chain/cronos#stateless-testcase -- gen /tmp/data/out
   --fullnodes 7 \
   --hostname_template "testplan-{index}" \
   --num_accounts 10 \
-  --num_txs 1000
+  --num_txs 1000 \
+  --num_max_workers 3000
```
Comment (yihuang) [inline testground/README.md:89]: more workers than accounts?

**Comment by yihuang** [comment]
> `3 7 testplan-{index} 200 15 3000`

what is the meaning of the fields.

**Comment by mmsqe** [comment]
> 3000

```
validator_count:   3
fullnode_count:    7
num_accounts:      200
num_txs:           15
num_max_workers:   3000
```

**Comment by yihuang** [comment]
> > 3000
> 
> ```
> validator_count:   3
> fullnode_count:    7
> num_accounts:      200
> num_txs:           15
> num_max_workers:   3000
> ```

what's the point of setting thread pool size larger than accounts, there's only one task for each account.

File: testground/benchmark/benchmark/sendtx.py [hunk 2]
```diff
@@ -77,12 +77,12 @@ def sendtx(w3: web3.Web3, acct: Account, tx_amount: int):
...
 
-def generate_load(cli, num_accounts, num_txs, **kwargs):
+def generate_load(cli, num_accounts, num_txs, num_max_workers, **kwargs):
     w3 = web3.Web3(web3.providers.HTTPProvider("http://localhost:8545"))
     assert w3.eth.chain_id == 777
     genesis_account = export_eth_account(cli, "account", **kwargs)
     accounts = fund_test_accounts(w3, genesis_account, num_accounts)
-    with ThreadPoolExecutor(max_workers=num_accounts) as executor:
+    with ThreadPoolExecutor(max_workers=num_max_workers) as executor:
         futs = (executor.submit(sendtx, w3, acct, num_txs) for acct in accounts)
```
Comment (mmsqe) [inline testground/benchmark/benchmark/sendtx.py:86]: task is based on num_max_workers, not based on account

Comment (yihuang) [reply testground/benchmark/benchmark/sendtx.py:86]: aren't we only `executor.submit` one task per account?

Comment (mmsqe) [reply testground/benchmark/benchmark/sendtx.py:86]: I see, it's only useful when less than num_accounts to avoid OOM

Comment (yihuang) [reply testground/benchmark/benchmark/sendtx.py:86]: yes, but the only reason to use too many num_accounts is only to improve concurrency level

File: testground/benchmark/benchmark/peer.py [hunk 3]
```diff
@@ -116,29 +116,9 @@ def gen_genesis(cli: ChainCommand, leader_home: Path, peers: List[PeerPacket]):
     )
 
 
-def patch_configs(home: Path, group: str, peers: str, block_executor: str):
+def patch_configs(home: Path, peers: str, config_patch, app_patch):
```
Comment (yihuang) [inline testground/benchmark/benchmark/peer.py:119]: it's also used in other places right now, better keep it backward compatible.

**Review by yihuang — APPROVED** [review]

**Pull request merged**